### PR TITLE
Entity classes updated to automate 6.3 Canned Role feature

### PR DESCRIPTION
**Pull request by jyejare**
Below two classes has been updated to automate 6.3 feature Canned Role:
1. Filter
2. Role

### CODE REVIEW

File: nailgun/entities.py [hunk 1]
```diff
@@ -2472,6 +2479,24 @@ def create_payload(self):
...
         return {u'filter': super(Filter, self).create_payload()}
 
+    def read(self, entity=None, attrs=None, ignore=None, params=None):
+        """Deal with different named data returned from the server
+        """
+        if ignore is None:
+            ignore = set()
+        if attrs is None:
+            attrs = self.read_json()
+            attrs['override'] = attrs.pop('override?')
```
Comment (svtkachenko) [inline nailgun/entities.py]: This 2 lines and 2 in the `else` scope looks identical. Maybe you can just move them outside of the `if` scope and remove `else` part?

**Comment by jyejare** [comment]
@svtkachenko , I have worked on your comment, please revisit.

**Comment by svtkachenko** [comment]
One last thing and coveralls will be happy :)
Add to `PathTestCase.test_nowhich` to test base path.

**Comment by jyejare** [comment]
@svtkachenko , Yeah I missed that. And Now I fixed it. I hope I didnt miss anything now.

File: nailgun/entities.py [hunk 2]
```diff
@@ -5546,14 +5566,62 @@ def __init__(self, server_config=None, **kwargs):
...
         super(Role, self).__init__(server_config, **kwargs)
 
+    def create_payload(self):
+        """Wrap submitted data within an extra dict.
+
+        For more information, see `Bugzilla #1151220
+        <https://bugzilla.redhat.com/show_bug.cgi?id=1151220>`_.
+
+        """
+        return {u'role': super(Role, self).create_payload()}
```
Comment (abalakh) [inline nailgun/entities.py]: Could you please provide some details why this change is needed? We didn't have extra dict for creating/updating the role and both operations were working for us. Or extra dict is needed for support of `location` & `organization` fields introduced in this PR?

Comment (jyejare) [reply nailgun/entities.py]: Thats true ... the extra dict is required for orgs and locations.

**Comment by jyejare** [comment]
Please refer following robotello PR for results:
https://github.com/SatelliteQE/robottelo/pull/5592 which is dependent on this PR.
